src/test/java/jenkins/branch/naming/ProjectNamingStrategyTest.java

```diff
@@ -0,0 +1,162 @@
+        ), pr);
+        stubGithubApiCall(String.format(
+            "/repos/%s/%s/pulls", MockGithubOrg.ORG_LOGIN, MockGithubRepository.REPO_NAME
+        ), Collections.singletonList(pr));
+
+        final MockGithubJenkinsfileContent jenkinsfileContent = new MockGithubJenkinsfileContent();
+        stubGithubApiCall(String.format(
+            "/repos/%s/%s/contents/", MockGithubOrg.ORG_LOGIN, MockGithubRepository.REPO_NAME
+        ), Collections.singletonList(jenkinsfileContent));
+
+        githubApiStub.start();
+    }
+
+    @AfterClass
+    public static void shutdownGithubApiStub() {
+        githubApiStub.stop(0);
+    }
+
+    @Test
+    @Issue("JENKINS-55348")
+    public void testRawStrategy() throws Exception {
+        testNamingStrategy(MultiBranchProjectDisplayNamingStrategy.RAW);
+    }
+
+    @Test
+    @Issue("JENKINS-55348")
+    public void testCompositeStrategy() throws Exception {
+        testNamingStrategy(MultiBranchProjectDisplayNamingStrategy.RAW_AND_OBJECT_DISPLAY_NAME);
+    }
+
+    @Test
+    @Issue("JENKINS-55348")
+    public void testObjectNameStrategy() throws Exception {
+        testNamingStrategy(MultiBranchProjectDisplayNamingStrategy.OBJECT_DISPLAY_NAME);
+    }
+
+    public void testNamingStrategy(final MultiBranchProjectDisplayNamingStrategy namingStrategy) throws Exception {
+        final Jenkins jenkinsInstance = r.jenkins;
+        final String projectName = String.format("Project_%s", namingStrategy.name());
+        final WorkflowMultiBranchProject multiBranchProject =
+            jenkinsInstance.createProject(WorkflowMultiBranchProject.class, projectName);
+
+        final GitHubSCMSource source = new GitHubSCMSource(MockGithubOrg.ORG_LOGIN, MockGithubRepository.REPO_NAME);
+        source.setApiUri(stubBaseUrl);
+        source.setTraits(Arrays.asList(
+            new OriginPullRequestDiscoveryTrait(2),
+            new MultiBranchProjectDisplayNamingTrait(namingStrategy)
+        ));
+        final BranchSource branchSource = new BranchSource(source);
+        multiBranchProject.getSourcesList().add(branchSource);
+        r.configRoundtrip(multiBranchProject);
+
+        final WorkflowJob prProject = jenkinsInstance.getItemByFullName(
+            String.format("%s/PR-%s", projectName, MockPullRequest.PR_NUMBER),
+            WorkflowJob.class
+        );
+
+        final String expectedName = namingStrategy.generateName(MockPullRequest.PR_NAME, MockPullRequest.PR_TITLE);
+        assertNotNull(prProject, "No job was created for the pull request");
+        assertEquals(expectedName, prProject.getDisplayName(), "The job name doesn't match the naming strategy");
+    }
+
+    private static int findAvailablePort() {
+        try (final ServerSocket serverSocket = new ServerSocket(0)) {
+            return serverSocket.getLocalPort();
+        } catch (final IOException e) {
+            throw new RuntimeException("Could not find an available port for GitHub API stub");
+        }
+    }
```

Comment:
this is a pure race condition and will lead to flaky tests.

the port is available at T1, but may not be at T2 when it is used by wiremock.

Rather get HttpServer to bind to any available random port (`0`) and then ask it which port it bound to.

Reply:
Indeed, fixed.